File: fundamentos/desafio-js/src/questions/get-next-id.ts
import { IAccount } from "../i-account";

/** Considerando que o id deve ser único e é sequencial, qual o próximo id possível para conta? */
export default function getNextId(...accounts: Array<IAccount>) {
  const nextId = accounts
    .sort((a, b) => b.id - a.id)[0].id + 1;

  console.log(`Considerando que o id deve ser único e é sequencial, qual o próximo id possível para conta? ${nextId}`);
}

File: fundamentos/desafio-dart/lib/questions/get_next_id.dart
import 'package:desafio_dart/domains/account.dart';

/// Considerando que o id deve ser único e é sequencial, qual o próximo id possível para conta?
void getNextId(List<Account> accounts) {
  final accs = accounts..sort((a, b) => b.id.compareTo(a.id));

  final nextId = accs.first.id + 1;

  print(
      'Considerando que o id deve ser único e é sequencial, qual o próximo id possível para conta? ${nextId}');
}
